Possibility to have xml file directly into module folder on OC 4.x

GeekoDev opened the pull request:
This modification gives the possibility to include an xml file directly inside oc 4.x extension folder, the file must be into /extension/[module]/vqmod/ in order to work.
This way a package can be fully integrated with its modification into the module package, to avoid extra operation of uploading vqmod file by FTP.

For this it was necessary to modify the vqmod.php file, which is kind of ugly compared to have only an xml file to upload, but unfortunately I see no other way to do that, as we can't vqmod the vqmod :)

GeekoDev committed:
Add files via upload

Make vqmod to also take xml files which are in /vqmod/ folder from each extension directory, this way once vqmod is installed we can load a module package with its vqmod file inside directly.

GeekoDev committed:
Update readme

JAY6390 commented:
If I'm understanding this correctly, you want to have an entire vqmod.php file in the OC install version? I'm thinking there might be a cleaner way of doing this with an additional file that can be added that has a list of path patterns to look for the other vqmod xml files. That way, vqmod.php could be upgraded independently of the OC version

GeekoDev commented:
You're right, I just did something to handle that in clean way directly by using pathReplaces file, I will submit the pull request in vqmod.

GeekoDev committed:
Delete vqmod.php

GeekoDev committed:
pathReplaces.php file to support OC4 extensions

pathReplaces.php file updated for latest vqmod update to support to include xml directly in OC4 module folder structure into extension/[mod]/vqmod/

GeekoDev committed:
Update README.md

GeekoDev committed:
Update README.md

GeekoDev commented:
Did it, see the pull request on main vqmod, I think it is quite clean this way, and we can use directly pathReplaces.php so no additional file necessary.

Sorry about README.md, I tried to restore original one but don't know I can remove the file from the pull request, if all ok for you please pull only the pathReplaces.php

JAY6390 commented:
Hi Geekodev

Thanks for that. I'm looking into creating an entirely new version of VQMod (3.0) with a couple of extra features. I'll include this as well as part of that release, and maybe a  If you'd like to be an early tester for it, send me an email (as per the emails in the copyright) and I'll let you know when I have a copy.

Kind regards
Jay

GeekoDev commented:
Hi Jay,

That's great, you can use this email: support@geekodev.com

dennisbuijsgmailcom commented:
Hi
Very good idea, was also thinking of how we can achieve this using the oc4 extension installer.

I like to test is also, support@openwebcreations.eu.
